src/SDKs/Azure.Core/data-plane/Azure.Core/ReliableStream.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    public static class ReliableStream
```

Comment:
I'm not sure that `Reliable` makes sense to me in this context.  Normally, when you see `Reliable` it indicates that once the call completes, the receiver has taken responsibility/ownership of the requested operation and guarantees it.   I think of things like "reliable messaging" where I can ask for this to be sent and the receiver ensures durability there;  if the process crashes immediately after that call returns, the message still gets sent either when the process restarts or by an intermediary.   It's a bit arbitrary, I know, but to me, this seems to fall short of that connotation.

Maybe we should consider something like `ResilientStream` or `RetriableStream` or similar?

Reply:
I like `RetriableStream`